src/main/scala/viper/gobra/frontend/Parser.scala

```diff
@@ -254,19 +271,131 @@ object Parser {
         // failed strategy application
         val updatedImports = rewrite(topdown(attempt(resolveImports)))(prog.imports)
-        val updatedProg = PProgram(prog.packageClause, prog.initPosts, updatedImports, prog.declarations)
-        pkg.positions.positions.dupPos(prog, updatedProg)
+        PProgram(prog.packageClause, prog.initPosts, updatedImports, prog.declarations).at(prog)
       })
       // create a new package node with the updated programs
-      val updatedPkg = PPackage(pkg.packageClause, updatedProgs, pkg.positions, pkg.info)
-      pkg.positions.positions.dupPos(pkg, updatedPkg)
+      val updatedPkg = PPackage(pkg.packageClause, updatedProgs, pkg.positions, pkg.info).at(pkg)
       // check whether an error has occurred
       if (failedNodes.isEmpty) Right(updatedPkg)
       else Left(failedNodes)
     }
   }
 
+  private class TerminationMeasurePostprocessor(override val positions: Positions, specOnly: Boolean) extends Postprocessor {
+    /**
+      * if `specOnly` is set to true, this postprocessor replaces all tuple termination measures specified for pure functions
+      * or pure methods by wildcard termination measures while maintaining their condition (if any). Note that termination
+      * measures specified for interface methods remain untouched.
+      * Furthermore, pure functions and pure methods are made abstract by turning their bodies into an additional
+      * postcondition.
+      *
+      * These steps ensure that imported pure functions and pure methods do not result in proof obligations as their
+      * postconditions and termination (in case a decreases measure has been provided) is simply assumed.
+      */
+    def postprocess(pkg: PPackage)(config: Config): Either[Vector[VerifierError], PPackage] = {
+      if (specOnly) replaceTerminationMeasures(pkg) else Right(pkg)
+    }
+
+    private def replaceTerminationMeasures(pkg: PPackage): Either[Vector[VerifierError], PPackage] = {
+      def createError(n: PNode, errorMsg: String): Vector[VerifierError] =
+        pkg.positions.translate(message(n, errorMsg), ParserError)
+
+      // unfortunately Kiama does not seem to offer a way to report errors while applying the strategy
+      // hence, we keep ourselves track of errors
+      var failedNodes: Vector[VerifierError] = Vector()
+
+      /**
+        * checks whether the body is either empty or consists of a single return statement and returns the corresponding expression
+        */
+      def getBodyExpr(body: Option[(PBodyParameterInfo, PBlock)]): Option[PExpression] = body match {
+        case Some((_, block)) =>
+          block.nonEmptyStmts match {
+            case Vector(PReturn(Vector(expr))) => Some(expr)
+            case _ =>
+              failedNodes = failedNodes ++ createError(block, s"the imported pure function's or method's body is expected to contain only a return statement")
+              None
+          }
+        case None => None
+      }
```

Comment:
Here and for `getOutParam` we duplicate checks, which add a bit of difficulty for maintenance as both these checks have to synced and might weaken in the future. I propose to add a comment to the other versions of these checks referencing the methods in this file, so that we do not forget that they are linked.